x/erc20/keeper/ibc_callbacks.go

```diff
@@ -74,9 +77,18 @@ func (k Keeper) OnRecvPacket(
 		return ack
 	}
 
-	// short-circuit: if the denom is not registered, conversion will fail
-	// so we can continue with the rest of the stack
-	if !k.IsDenomRegistered(ctx, coin.Denom) {
+	pairID := k.GetTokenPairID(ctx, coin.Denom)
+	if len(pairID) == 0 {
+		// short-circuit: if the denom is not registered, conversion will fail
+		// so we can continue with the rest of the stack
+		return ack
+	}
+
+	pair, _ := k.GetTokenPair(ctx, pairID)
+
+	if !pair.Enabled {
+		// FIXME: should I return ack/error ?
```

Comment:
@fedekunze do you agree, that if the token pair is disabled, the IBC transfer is still successful?